doc/server.py

```diff
@@ -0,0 +1,25 @@
+from http.server import HTTPServer, SimpleHTTPRequestHandler
+from subprocess import call
+
+PORT = 8000
+URL = f"http://localhost:{PORT}"
+
+
+class AutoBrowseHTTPServer(HTTPServer):
+    def server_activate(self):
+        HTTPServer.server_activate(self)
+        print(f"Serving documentation at {URL}")
+        call(["xdg-open", URL])
+
+
+class UncachedHTTPHandler(SimpleHTTPRequestHandler):
+    def end_headers(self):
+        self.send_header("Cache-Control", "no-cache, no-store, must-revalidate")
+        self.send_header("Pragma", "no-cache")
+        self.send_header("Expires", "0")
+        super().end_headers()
```

Comment:
> > Adds a HTTP server for testing the docs locally. This works better than directly browsing the filesystem; for example cookies can be used, so the site "theme" is remembered.
> 
> IMHO, it would be better without caching to ensure reproducibiliy when "testing" `nix build .#docs`. However, I am also fine with some caching.

That is what this code is doing; disabling browser cache.

Reply:
> > > Adds a HTTP server for testing the docs locally. This works better than directly browsing the filesystem; for example cookies can be used, so the site "theme" is remembered.
> >
> > IMHO, it would be better without caching to ensure reproducibiliy when "testing" `nix build .#docs`. However, I am also fine with some caching.
>
> That is what this code is doing; disabling browser cache.

I mentioned this in the patch notes:

> ```patch
> Caching can be configured with http-server's '-c' option. Currently, caching is
> disabled with '-c-1':
>
>     http-server ${docs} --port 0 -c-1 -o
> ```